docs/nima/testing.adoc

```diff
@@ -0,0 +1,397 @@
+
+            assertThat(response.status(), is(Http.Status.OK_200));  <6>
+        }
+    }
+}
+----
+<1> Use `@ServerTest` to trigger the testing framework;
+<2> Inject `Http1Client` for the test;
+<3> SetUp routing for the test;
+<4> Regular `JUnit` test method;
+<5> Call the `client` to obtain server response;
+<6> Perform necessary assertions;
+
+To trigger the framework starting and configuring Níma server, the testing class should be annotated with `@ServerTest` annotation.
+
+In this test, the `Http1Client` is used, which means that the framework will take care of creating, configuring, and injecting this object as a parameter to the constructor.
+
+To set up routing, a static method annotated with `@SetUpRoute` is present. The framework uses this method to inject the configured routing to the subject of testing – in the current case `Quickstart` application.
+
+As everything above is performed by the testing framework, regular unit tests can be done. When all tests are completed, the testing framework will shut down the Níma server.
+
+=== Routing tests
+
+If there is no need to set up and run a server, a `DirectClient` can be used. It is a testing client that bypasses HTTP transport and directly invokes the router.
+
+.Routing test using `@RoutingTest` and `DirectClient`.
+[source, java]
+----
+@RoutingTest    <1>
+class RoutingTest {
+
+    private final Http1Client client;
+
+    protected RoutingTest(DirectClient client) { <2>
+        this.client = client;
+    }
+
+    @SetUpRoute <3>
+    static void routing(HttpRouting.Builder builder) {
+        QuickstartMain.routing(builder);
+    }
+
+    @Test <4>
+    void testRootRoute() {
+        try (Http1ClientResponse response = client.get("/greet")
+                .request()) {   <5>
+
+            JsonObject json = response.as(JsonObject.class);    <6>
+            assertThat(json.getString("message"), is("Hello World!"));
+        }
+    }
+}
+----
+<1> Use `@RoutingTest` to trigger the testing framework;
+<2> Inject `DirectClient` for the test;
+<3> SetUp routing for the test;
+<4> A regular `JUnit` test method;
+<5> Call the `client` to obtain server response;
+<6> Perform necessary assertions;
+
+If only routing tests are required, this is a "lighter" way of testing since the framework will not configure and run the full Níma server. This way, no real ports will be opened. All the communication will be done through `DirectClient` which makes the tests very effective.
+
+It is required to annotate the test class with `@RoutingTest` annotation to trigger the server to do the configuration. Thus, it will inject the `DirectClient`, which can then be used in unit tests.
+
+Routing is configured the same way as in full server testing using the `@SetUpRoute` annotation.
+
+== Additional Information
+
+=== WebSocket Testing
+
+In case WebSocket testing is required, there is an additional module for this. It is necessary to include the following Maven dependency to Project's pom file:
```

Comment:
If WebSocket testing is required, there is an additional module for it. It is necessary to include the following Maven dependency to the Project's pom file: